src/v/cloud_storage/cache_service.cc

```diff
@@ -1644,4 +1672,17 @@ ss::future<> cache::initialize(std::filesystem::path cache_dir) {
     }
 }
 
+uint64_t cache::trim_target_bytes(std::optional<uint64_t> size_limit_override) {
+    // Pick the override, the manually configured threshold, or 80% of max, in
+    // that order.
+    return size_limit_override.value_or(
+      _trim_threshold_bytes().value_or(_max_bytes * 80 / 100));
```

Comment:
this makes the feature always on
can we use the new early trim in the background only if the config is not nullopt? we're introducing a lot of new behaviors into the cache, it'd be great to be able to enable all this stuff gradually